app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java

```diff
@@ -94,6 +90,46 @@ private void executeGrpcTask(GrpcTask task) {
         mMoveForwardButton.setEnabled(false);
```

Comment:
Since the buttons are re-enabled in postExecute, wouldn't it be fair to disable them in preExecute?

Alternatively, if you don't want them disabled asynchronously, how about you give the task a callback which would enable the buttons?

Ideally, it should be the same method e.g. `enableButtons(bool)` which goes through all the buttons, to avoid code duplicates and its consequences.